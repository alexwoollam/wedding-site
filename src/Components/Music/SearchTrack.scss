@import '../../variables';

.track-selector{
  & .btn-submit{
    border: 2px solid $dustygreen;
    color: $dustygreen;
  }
}