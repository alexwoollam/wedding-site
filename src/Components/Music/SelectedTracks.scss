@import '../../variables';

.btn-remove{
  background-color: $dustypink;
  border: 2px solid $dustypink--dark;
  &:hover{
    background-color: $dustypink--dark;
  }
}
