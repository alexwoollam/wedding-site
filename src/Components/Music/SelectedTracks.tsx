import React from 'react';
import {Container, Row, Col, Button} from 'reactstrap';

interface PropsDefined {
    data: any,
    addTrack?: any,
    removeTrack?: any,
    isSelected?: boolean,
}

function SelectedTracks( props: PropsDefined ){

    if( ! props.data ){ 
        return null;
    }

    let items = props.data;

    if( items.length > 0 ){
        return (
            <Container className={ 'mb-3' }>
                {items.map( (track, index)=>(
                    <Row key={ index } style={{border: '2px dashed lightgrey'}} className='p-3 mt-2 rounded'>
                        <Col md={1}>
                            <img className={'rounded'} src={ track.trackimage } alt={"artwork for " + track.trackname } />
                        </Col>
                        <Col>
                            <h4>{ track.trackname }</h4>
                            <h5>{ track.trackartist }</h5>
                        </Col>
                        <Col md={1} className="d-flex justify-content-center">
                            <Button data-testid='remove-button' className={'btn-danger'} onClick={ () => props.removeTrack(track.id) }>Remove</Button>
                        </Col>
                    </Row>
                ))}
            </Container>
        );
    } else {
        return null;
    }
} 

export default SelectedTracks;