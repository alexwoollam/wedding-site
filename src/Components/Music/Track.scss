@import '../../variables';

.track-card{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  min-height: 540px;
  &-button{
    background-color: $dustygreen;
    &:hover{
      background-color: $dustygreen--dark;
    }
  }
  &-body{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding: 0px;
    min-width: 100%;

    & .track-card-preview{
      align-self: end;
      min-width: 100%;
    }
  }
}

.cover-preview{
  width: 280px;
  height: 280px;
  border-radius: 5px;

}