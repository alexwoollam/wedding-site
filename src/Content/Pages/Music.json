{
  "title": "Music",
  "subtitle": "What gets you dancing ( or awkwardly swaying )",
  "intro": "we're currently curating a playlist for the evening's proceedings, please add your suggestions."
}