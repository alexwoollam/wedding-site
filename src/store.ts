import { configureStore } from '@reduxjs/toolkit'
import rsvpReducer from "./Reducers/rsvpReducer";
import tracksReducer from "./Reducers/tracksReducer";

export const store = configureStore({
    reducer: {
        rsvp: rsvpReducer,
        tracks: tracksReducer
    },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
