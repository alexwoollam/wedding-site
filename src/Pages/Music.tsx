import React from 'react';
import Heading from '../Components/Text/Heading';
import {SearchTracks} from '../Components/Music/SearchTrack';
import { Row, Col } from 'reactstrap';
import Content from "../Content/Pages/Music.json";
import TextBlock from "../Components/Text/TextBlock";
import Page from "../Components/Layout/Page";
import Bugs from "../Components/Bugs";

function Music() {
  return (
      <Page>
          <div className={"section"} style={{ marginBottom: '120px'}}>
              <Row>
                  <Col>
                      <Bugs/>
                      <Heading main subtitle={Content.subtitle}>{Content.title}</Heading>
                      <TextBlock>{Content.intro}</TextBlock>
                  </Col>
              </Row>
              <Row className={"w-100"}>
                  <Col className={"col-12"}>
                      <SearchTracks/>
                  </Col>
              </Row>
          </div>
      </Page>
  );
}

export default Music;