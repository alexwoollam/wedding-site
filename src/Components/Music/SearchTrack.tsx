// @ts-nocheck
import React, { Component } from 'react';
import SpotifyWebApi from 'spotify-web-api-js';
import SpotifyAuthApi from 'spotify-web-api-node';
import Track from './Track';
import SelectedTracks from './SelectedTracks';
import { Col, Form, Input, Button } from 'reactstrap';
import { Submit } from '../Forms/KeepMeUpdated';

interface theTracksInterface{
    tracks: object,
}

interface trackListInterface{
    id: number, 
    trackid: string,
    trackname: string,
    trackartist: string,
    trackimage: string,
}

class SearchTracks extends Component{
    
    constructor(props) {
        super(props);

        this.state = {
            theToken: [],
            spotifyCreds: true,
            trackList: [],
            theTracks: {
                tracks: []
            },
        };

        this.searchForTrack = this.searchForTrack.bind(this);
        this.addTrack = this.addTrack.bind(this);
        this.removeTrack = this.removeTrack.bind(this);
    }

    addTrack( id: string, title: string, artist: string, image: string ){
        var the_length;
        if(this.state.trackList){
            the_length = this.state.trackList.length; 
        } else {
            the_length = 0;
        }

        const object = {
            id: the_length, 
            trackid: id,
            trackname: title,
            trackartist: artist,
            trackimage: image,
        }
        console.log(object);
        console.log('^^object');
        var joined = this.state.trackList.concat(object);
        this.setState({ trackList: joined });

    }

    removeTrack( id: number ){
        const list = this.state.trackList.filter(trackList => trackList.id !== id);
        this.setState({ trackList: list });
    }

    componentDidMount(){
        const CLIENT_ID:string = process.env.REACT_APP_SPOTIFYCLI!;
        const CLIENT_SEC:string = process.env.REACT_APP_SPOTIFYSEC!;

        var token = btoa( CLIENT_ID + ':' + CLIENT_SEC );

        fetch("https://accounts.spotify.com/api/token", {
        body: "grant_type=client_credentials",
        headers: {
            'Authorization': 'Basic ' + token,
            'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
        },
        method: "POST"
        })
        .then(response => response.json())
        .then( data => 
            this.setState({
                theToken: data.access_token
            })
        );
    };

    submitTracks() {
        
        let trackArray: string[] = [];
        this.state.trackList.map((track: any) => (
            trackArray.push('spotify:track:'+track.trackid)
        ));
        /**
         * yeeah, we're going to have to dump these to a google sheet or something.
         */

    }

    searchForTrack( search:any ){
        var spotifyApi = new SpotifyWebApi();
        spotifyApi.setAccessToken(this.state.theToken);
        spotifyApi.searchTracks(search.target.value).then(
            (data) => {        
                this.setState({
                    theTracks: data,
                })
            },
        );
        

        
    }

    render() {
        return (
            <Col
                className='m-auto'
            >
                <Button data-testid='submit-button' onClick={ this.submitTracks }>Lets rock!</Button>
                <SelectedTracks data={ this.state.trackList } removeTrack={ this.removeTrack } isSelected={true} />
                <Form className="m-auto max-w-md my-10">
                    <Input 
                    type="text"
                    id="search-track"
                    data-testid='search-input'
                    className="p-3 block w-full rounded-md bg-gray-100 border-transparent focus:border-gray-500 focus:bg-white focus:ring-0"
                    placeholder="Song title"
                    onChange={ this.searchForTrack }
                    ></Input>
                </Form>
                <Track data={ this.state.theTracks.tracks } addTrack={ this.addTrack } />
            </Col>
        );
    };
    
}

export default SearchTracks;