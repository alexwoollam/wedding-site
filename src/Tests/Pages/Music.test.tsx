import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Music from '../../Pages/Music';
import {Provider} from "react-redux";
import {store} from "../../store";

describe('<Music /> tests.', () => {
    test('Music page should render', () => {
        render(<Provider store={store}><Music /></Provider>);
        const title = screen.getByText('Music');
        expect(title).toBeInTheDocument();
    });
    it('should renders correctly', () => {
        const tree = renderer
            .create(<Provider store={store}><Music /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
