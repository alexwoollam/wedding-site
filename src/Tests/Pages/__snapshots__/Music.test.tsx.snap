// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Music /> tests. should renders correctly 1`] = `
<div
  className="App container mx-auto row"
>
  <div
    className="Intro col"
  >
    <h1
      className="text-grey-700 font-serif text-3xl text-center m-2"
      data-testid="text-block"
    >
      What music would you dance too?
    </h1>
    <div
      className="m-auto col"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Lets rock!
      </button>
      <div
        className="mb-3 container"
      />
      <form
        className="m-auto max-w-md my-10"
      >
        <input
          className="p-3 block w-full rounded-md bg-gray-100 border-transparent focus:border-gray-500 focus:bg-white focus:ring-0 form-control"
          onChange={[Function]}
          placeholder="Song title"
          type="text"
        />
      </form>
    </div>
  </div>
</div>
`;
